.github/workflows/run_release.yaml

```diff
@@ -8,22 +8,22 @@ on:
     tags-ignore:
       - "**" # Ignore all tags to prevent duplicate builds when tags are pushed.
 
-  # Trigger a stable version release when a GitHub release is published.
-  release:
-    types: [published]
-
-  # Allows manual workflow runs from GitHub's UI, with the ability to specify the type of release.
+  # Trigger a stable version release via GitHub's UI, with the ability to specify the type of release.
   workflow_dispatch:
     inputs:
       release_type:
         description: Release type
         required: true
         type: choice
-        default: alpha
+        default: auto
         options:
-          - alpha
-          - beta
-          - final
+          - auto
+          - custom
```

Comment:
i would like to be able to trigger major/minor/patch specifically (just like we have it in the js crawlee) instead of relying on the git messages. some feature commits are not really worth a feature release (but you want them in the changelog and its just wrong to call them fixes just because of this)

Reply:
...it looks like `poetry version` could also work for us.

Reply:
How is this about git cliff? You use that to generate changelog between two git tags or commit hashes I'd guess, or its not working that way? The option I am after would only let you decide the version bump manually instead of relying on the git messages.

Reply:
[This is how](https://github.com/apify/crawlee-python/pull/330/files#diff-4992bcb5677b9499cbec92fcb2bc14c4e0232be0e127673730e7a6582e594c3dR60-R66) - it has a flag that outputs the new semver based on commits since the last tag. I used it because it was handy. The `workflow_dispatch` can also accept a custom version number, which will be used directly.

Reply:
> Also, kinda off topic, but some people (such as @netmilk IIRC) would tell you that attaching emotional value to version numbers is wrong and you should just follow the semantic versioning spec like a robot. Every side of the argument has its points IMO.

It's not emotional value, if you change something that has no effect on the user facing API (and therefore on docs), that is not a feature release to me. Yet it could be a feature commit (think of a small improvement of something). Also being able to market feature releases is a good thing.

For me this is about being in control, not about emotions. If you commit one thing with a wrong prefix (and that happens all the time, remember the small breaking change with URLs from yesterday?), you are stuck with what you have in master.

Reply:
I agree with @B4nan on this.

Reply:
> You can already set a custom version, can't that be used to achieve the same thing you could do with a patch/minor/major toggle?

Yes, it can be used for the same, but is less ergonomic. It means you need to check the version you are on, instead of just releasing the next bump based on the type. You add another possibility for the user to screw it up. I want something in between, not full auto, but not full custom. We are using this for quite some time in crawlee, and it feels really nice, so I would like to see it the same way - especially because we want to port this setup to other repositories in the long run.

> While we're at the topic, can we not fix incorrect commit types by push-forcing to master?

Force push to master is a big no-no, especially in open-source repositories. This can piss all the other maintainers off easily, seen that many times in nette community, as the author liked to do this a lot. You break their local versions since you change commit hashes.

There are exceptions to everything, but I can justify force pushing only for the very last commit done by yourself, within very short time window after they were pushed (lets say minutes).

Reply:
> Yes, it can be used for the same, but is less ergonomic. It means you need to check the version you are on, instead of just releasing the next bump based on the type. You add another possibility for the user to screw it up. I want something in between, not full auto, but not full custom. We are using this for quite some time in crawlee, and it feels really nice, so I would like to see it the same way - especially because we want to port this setup to other repositories in the long run.

Sure, I'm in favor of that, but I'd prefer to do that after this PR is merged and tested. Autogenerated changelogs and triggering releases via workflow_dispatch are enough of a milestone IMO.

> Force push to master is a big no-no, especially in open-source repositories. This can piss all the other maintainers collaborators off easily, seen that many times in nette community, as the author liked to do this a lot. You break their local versions since you change commit hashes.

Haha, I remember that. I agree that the negative impact on contributors is not worth it.

Reply:
Ok, I am fine merging this without it, that's fair. But I would like to see it implemented before next release.